# HG changeset patch
# Parent  5ab3699200d68360a9398e63f2beb88afd3d84b2
Fix version detection and improve hardening flags in the Hombrew formulae

diff --git a/dist/homebrew/lightsd.rb b/dist/homebrew/lightsd.rb
--- a/dist/homebrew/lightsd.rb
+++ b/dist/homebrew/lightsd.rb
@@ -5,6 +5,8 @@
   homepage "https://github.com/lopter/lightsd/"
   url "{{ archive_url  }}"
   sha256 "{{ archive_sha256 }}"
+  # This will have ~ instead of - for rc versions:
+  version "{{ version|replace("-", "~") }}
   revision {{ build_number }}
 
   depends_on "cmake" => :build
@@ -22,7 +24,7 @@
     # proper release flags:
     cflags = %W[
       -fstack-protector-strong
-      --param=ssp-buffer-size=4
+      -D_FORTIFY_SOURCE=2
       -O3
       -DNDEBUG
     ]
@@ -69,8 +71,9 @@
   end
 
   def caveats; <<-EOS.undent
-    Once you've started lightsd with launchctl load (see below), you can start
-    poking around with lightsc.py:
+    Once you've started lightsd (see the brew services command below, if you're
+    using tmux, remember that you will need to run it outside tmux), you can
+    start poking around with lightsc.py:
 
       `lightsd --prefix`/share/lightsd/examples/lightsc.py
     EOS
